event_model/__init__.py

```diff
@@ -2165,21 +2171,23 @@ def compose_run(*, uid=None, time=None, metadata=None, validate=True):
     # Define some mutable state to be shared internally by the closures composed
     # below.
     streams = {}
```

Comment:
```suggestion
    streams = {}
    if event_counters is None:
        event_counters = {}
```

Reply:
I prefer the or syntax for this, though it would require changing the others and I'm not sure which nsls prefer? @danielballan 

For now I'll use and if `is None` and we can change it another time if need be.